lib/puppet_x/chocolatey/chocolatey_common.rb

```diff
@@ -0,0 +1,81 @@
+require 'pathname'
+require Pathname.new(__FILE__).dirname + 'chocolatey_version'
+require Pathname.new(__FILE__).dirname + 'chocolatey_install'
+
+module PuppetX
+  module Chocolatey
+    module ChocolateyCommon
+
+      def file_exists?(path)
+        File.exist?(path)
+      end
+      module_function :file_exists?
+
+      def chocolatey_command
+        if Puppet::Util::Platform.windows?
+          chocoInstallPath = PuppetX::Chocolatey::ChocolateyInstall.install_path
+          #default_location = $::choco_installpath || ENV['ALLUSERSPROFILE'] + '\chocolatey'
+          chocopath = ('C:\ProgramData\chocolatey' if file_exists?('C:\ProgramData\chocolatey\bin\choco.exe')) ||
+              (chocoInstallPath if chocoInstallPath && file_exists?("#{chocoInstallPath}\\bin\\choco.exe")) ||
+              (ENV['ChocolateyInstall'] if ENV['ChocolateyInstall'] && file_exists?("#{ENV['ChocolateyInstall']}\\bin\\choco.exe")) ||
+              ('C:\Chocolatey' if file_exists?('C:\Chocolatey\bin\choco.exe')) ||
+              "#{ENV['ALLUSERSPROFILE']}\\chocolatey"
+
+          chocopath += '\bin\choco.exe'
+        else
+          chocopath = 'choco.exe'
+        end
+
+        chocopath
+      end
+      module_function :chocolatey_command
+
+      def set_env_chocolateyinstall
```

Comment:
No, not at all. The plan is to convert the package code over to using this. It was getting a little out of scope for this ticket as it was.